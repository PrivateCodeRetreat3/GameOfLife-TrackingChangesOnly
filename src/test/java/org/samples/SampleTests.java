package org.samples;


import org.approvaltests.Approvals;
import org.approvaltests.StoryBoard;
import org.junit.jupiter.api.Test;

import java.awt.*;

public class SampleTests {
    @Test
    public void testSingleCellAlive() {
        StoryBoard b = new StoryBoard();
        GameOfLife g = new GameOfLife(new Point(1, 1));
        b.add(g.toString());
        g.advance();
        b.add(g.toString());
        Approvals.verify(b);
    }

    @Test
    public void testGlider() {
        StoryBoard b = new StoryBoard();
        GameOfLife g = new GameOfLife(new Point(1, 1),
                 new Point(2, 2),
                 new Point(3, 0),
                 new Point(3, 1),
                 new Point(3, 2)
        );
        b.add(g.toString());
        for (int i = 0; i < 4; i++) {
            g.advance();
            b.add(g.toString());
        }
        Approvals.verify(b);
    }
}
