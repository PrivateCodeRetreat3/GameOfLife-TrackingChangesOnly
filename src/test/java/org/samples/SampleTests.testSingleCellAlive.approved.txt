Initial:
_ _ _ _ _ _ _ _ _ _ 
_ X _ _ _ _ _ _ _ _ 
_ _ _ _ _ _ _ _ _ _ 
_ _ _ _ _ _ _ _ _ _ 
_ _ _ _ _ _ _ _ _ _ 
_ _ _ _ _ _ _ _ _ _ 
_ _ _ _ _ _ _ _ _ _ 
_ _ _ _ _ _ _ _ _ _ 
_ _ _ _ _ _ _ _ _ _ 
_ _ _ _ _ _ _ _ _ _ 


Frame #1:
_ _ _ _ _ _ _ _ _ _ 
_ _ _ _ _ _ _ _ _ _ 
_ _ _ _ _ _ _ _ _ _ 
_ _ _ _ _ _ _ _ _ _ 
_ _ _ _ _ _ _ _ _ _ 
_ _ _ _ _ _ _ _ _ _ 
_ _ _ _ _ _ _ _ _ _ 
_ _ _ _ _ _ _ _ _ _ 
_ _ _ _ _ _ _ _ _ _ 
_ _ _ _ _ _ _ _ _ _ 
